cpp/src/arrow/flight/client.h

```diff
@@ -65,6 +65,9 @@ class ARROW_FLIGHT_EXPORT FlightCallOptions {
 
   /// \brief IPC writer options, if applicable for the call.
   ipc::IpcWriteOptions write_options;
+
+  /// \brief Metadata for client to add to context.
+  std::vector<std::pair<std::string, std::string>> metadata;
```

Comment:
I believe we call it headers elsewhere, so this should stay consistent with that.

Reply:
Will correct this.